src/Boundaries/Sediments/simple_multi_G.jl

```diff
@@ -138,30 +139,37 @@ adapt_structure(to, sediment::SimpleMultiG) =
-                              biogeochemical_drift_velocity(bgc, Val(:bPOC)).w[i, j, 1] * tracers.bPOC[i, j, 1])
+
+        carbon_deposition = carbon_flux(grid, advection, bgc, tracers, i, j) * Δz
                         
-        nitrogen_deposition = -(biogeochemical_drift_velocity(bgc, Val(:sPON)).w[i, j, 1] * tracers.sPON[i, j, 1] +
-                                biogeochemical_drift_velocity(bgc, Val(:bPON)).w[i, j, 1] * tracers.bPON[i, j, 1])
+        nitrogen_deposition = nitrogen_flux(grid, advection, bgc, tracers, i, j) * Δz
 
         # rates
-        Cᵐⁱⁿ = sediment.fields.C_slow[i, j, 1] * sediment.slow_decay_rate + sediment.fields.C_fast[i, j, 1] * sediment.fast_decay_rate
-        Nᵐⁱⁿ = sediment.fields.N_slow[i, j, 1] * sediment.slow_decay_rate + sediment.fields.N_fast[i, j, 1] * sediment.fast_decay_rate
+        C_min_slow = sediment.fields.C_slow[i, j, 1] * sediment.slow_decay_rate
+        C_min_fast = sediment.fields.C_fast[i, j, 1] * sediment.fast_decay_rate
+
+        N_min_slow = sediment.fields.N_slow[i, j, 1] * sediment.slow_decay_rate
+        N_min_fast = sediment.fields.N_fast[i, j, 1] * sediment.fast_decay_rate
+
+        Cᵐⁱⁿ = C_min_slow + C_min_fast
+        Nᵐⁱⁿ = N_min_slow + N_min_fast
         
         k = Cᵐⁱⁿ * day / (sediment.fields.C_slow[i, j, 1] + sediment.fields.C_fast[i, j, 1])
 
         # sediment evolution
-        sediment.tendencies.Gⁿ.C_slow[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.slow_fraction * carbon_deposition - sediment.slow_decay_rate * sediment.fields.C_slow[i, j, 1]
-        sediment.tendencies.Gⁿ.C_fast[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.fast_fraction * carbon_deposition - sediment.slow_decay_rate * sediment.fields.C_fast[i, j, 1]
-        sediment.tendencies.Gⁿ.C_ref[i, j, 1] = max(0.0, sediment.refactory_fraction * carbon_deposition)
+        sediment.tendencies.Gⁿ.C_slow[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.slow_fraction * carbon_deposition - C_min_slow
+        sediment.tendencies.Gⁿ.C_fast[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.fast_fraction * carbon_deposition - C_min_fast
+        sediment.tendencies.Gⁿ.C_ref[i, j, 1] = sediment.refactory_fraction * carbon_deposition
 
-        sediment.tendencies.Gⁿ.N_slow[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.slow_fraction * nitrogen_deposition - sediment.slow_decay_rate * sediment.fields.N_slow[i, j, 1]
-        sediment.tendencies.Gⁿ.N_fast[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.fast_fraction * nitrogen_deposition - sediment.slow_decay_rate * sediment.fields.N_fast[i, j, 1]
-        sediment.tendencies.Gⁿ.N_ref[i, j, 1] = max(0.0, sediment.refactory_fraction * nitrogen_deposition)
+        sediment.tendencies.Gⁿ.N_slow[i, j, 1] = (1 - sediment.refactory_fraction) * sediment.slow_fraction * nitrogen_deposition - N_min_slow
```

Comment:
inbounds?

Reply:
Also to the lines below

Reply:
I think this whole block is inboundsed right?

Reply:
Yes! I missed it!